File: 132P/primes.go
package main
import "fmt"

const SIZE int = 500 // table size
var prime [SIZE]int  // primes array

func isprime(n int, j int) bool {
  for k := 1; k < j; k++ {
    p := prime[k]
    if n % p == 0 { return false }
    if n / p <= p { break }
  }
  return true
}
func spit() {
  const LINEZ int = 50
  for i := 0; i < LINEZ; i++ {
    buf := ""
    for j := i; j < SIZE; j += LINEZ {
      buf += fmt.Sprintf(" %04d", prime[j])
    }
    fmt.Println(buf)
  }
}
func main() {
  fmt.Printf("Table of first 500 primes\n")
  prime[0] = 2
  for j, n := 1, 3;; n += 2 {
    if isprime(n, j) {
      prime[j] = n
      j++
      if j == SIZE { break }
    }
  }
  spit()
}
// log:

File: 132P/primes.cpp
#include <cstdio>
#include <string>
#include <vector>
using namespace std;

typedef vector<int> vint_t;

string fmt(int n)
{
  static const int w = 4;
  static const string z{"000"};  
  string s = z + to_string(n);
  return " " + s.substr(s.length() - w, w);
}
bool is_prime(vint_t& p, int n)
{
  for (int k = 1;; ++k) {
    if (n%p[k] == 0   ) return false;
    if (n/p[k] <= p[k]) return true ;
  }
}
vint_t get_primes(int size)
{
  vint_t p{2, 3};
  
  for (int n = 5; p.size() < size; n += 2) {
    if (is_prime(p, n)) p.push_back(n);
  }
  return p;
}
void spit(const vint_t& p, int linez)
{
  printf("Table of first %lu primes:\n", p.size());
  for (int i = 0; i < linez; ++i) {
    string s;
    for (int j = i; j < p.size(); j += linez) {
      s += fmt(p[j]);
    }
    puts(s.c_str());
  }
}
int main()
{
  spit(get_primes(500), 50);
}
// log:
